From 172bb2ef2489d4a0129827837cc91f8e4f55601b Mon Sep 17 00:00:00 2001
Message-Id: <172bb2ef2489d4a0129827837cc91f8e4f55601b.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Fri, 6 Sep 2013 19:47:32 +0530
Subject: [PATCH 130/312] FOR_UPSTREAM [VPG]: drm/i915: Program Valleyview DP
 M/N registers in case eDP as well

With this change BYT DP M/N registers are programmed. This was needed
to stabilze dual display feature. Without this change HDMI doesn't get
enabled if connected during boot with EFI. Sporadic behaviour with HDMI
hotplug was observed before which is no more reproducible with this
patch.

Issue: APDEV-1039
OTC-Tracker: VIZ-3441
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>
Change-Id: Ibe168f3bfa13f1673c1d6e1a3fef8b5fd1035e86
Reviewed-on: https://vpg-git.iind.intel.com/440
Reviewed-by: Kumar, Shobhit <shobhit.kumar@intel.com>
Tested-by: Kumar, Shobhit <shobhit.kumar@intel.com>
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index d48d3f9..e102db1 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -4539,7 +4539,8 @@ static void ironlake_crtc_enable(struct drm_crtc *crtc)
 	if (intel_crtc->config.has_pch_encoder)
 		intel_prepare_shared_dpll(intel_crtc);
 
-	if (intel_crtc->config.has_dp_encoder)
+	if (intel_crtc->config.has_dp_encoder ||
+		intel_pipe_has_type(&intel_crtc->base, INTEL_OUTPUT_EDP))
 		intel_dp_set_m_n(intel_crtc);
 
 	intel_set_pipe_timings(intel_crtc);
@@ -4651,7 +4652,8 @@ static void haswell_crtc_enable(struct drm_crtc *crtc)
 	if (intel_crtc->active)
 		return;
 
-	if (intel_crtc->config.has_dp_encoder)
+	if (intel_crtc->config.has_dp_encoder ||
+		intel_pipe_has_type(&intel_crtc->base, INTEL_OUTPUT_EDP))
 		intel_dp_set_m_n(intel_crtc);
 
 	intel_set_pipe_timings(intel_crtc);
@@ -5181,7 +5183,8 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 	/* Set up the display plane register */
 	dspcntr = DISPPLANE_GAMMA_ENABLE;
 
-	if (intel_crtc->config.has_dp_encoder)
+	if (intel_crtc->config.has_dp_encoder ||
+		intel_pipe_has_type(&intel_crtc->base, INTEL_OUTPUT_EDP))
 		intel_dp_set_m_n(intel_crtc);
 
 	intel_set_pipe_timings(intel_crtc);
@@ -5286,7 +5289,8 @@ static void i9xx_crtc_enable(struct drm_crtc *crtc)
 	else
 		dspcntr |= DISPPLANE_SEL_PIPE_B;
 
-	if (intel_crtc->config.has_dp_encoder)
+	if (intel_crtc->config.has_dp_encoder ||
+		intel_pipe_has_type(&intel_crtc->base, INTEL_OUTPUT_EDP))
 		intel_dp_set_m_n(intel_crtc);
 
 	intel_set_pipe_timings(intel_crtc);
-- 
1.7.9.5

